src/main/io/osd.c

```diff
@@ -1184,6 +1209,58 @@ static bool osdDrawSingleElement(uint8_t item)
     return true;
 }
 
+#ifdef USE_OSD_STICK_OVERLAY
+static void osdDrawStickOverlayAxis(uint8_t xpos, uint8_t ypos)
+{
+
+    for (unsigned x = 0; x < OSD_STICK_OVERLAY_WIDTH; x++) {
+        for (unsigned  y = 0; y < OSD_STICK_OVERLAY_HEIGHT; y++) {
+            // draw the axes, vertical and horizonal
+            if ((x == ((OSD_STICK_OVERLAY_WIDTH-1)/2)) && (y == (OSD_STICK_OVERLAY_HEIGHT-1)/2)) {
+                displayWriteChar(osdDisplayPort, xpos+x, ypos+y, STICK_OVERLAY_CROSS_CHAR);
+            } else if (x == ((OSD_STICK_OVERLAY_WIDTH-1)/2)) {
+                displayWriteChar(osdDisplayPort, xpos+x, ypos+y, STICK_OVERLAY_VERTICAL_CHAR);
+            } else if (y == ((OSD_STICK_OVERLAY_HEIGHT-1)/2)) {
+                displayWriteChar(osdDisplayPort, xpos+x, ypos+y, STICK_OVERLAY_HORIZONTAL_CHAR);
+            }
+        }
+    }
+}
+
+static void osdDrawStickOverlayAxisItem(osd_items_e osd_item)
+{
+    osdDrawStickOverlayAxis(OSD_X(osdConfig()->item_pos[osd_item]),
+                            OSD_Y(osdConfig()->item_pos[osd_item]));
+}
+
+static void osdDrawStickOverlayPos(osd_items_e osd_item, uint8_t xpos, uint8_t ypos)
+{
+
+    uint8_t elemPosX = OSD_X(osdConfig()->item_pos[osd_item]);
+    uint8_t elemPosY = OSD_Y(osdConfig()->item_pos[osd_item]);
+
+    displayWriteChar(osdDisplayPort, elemPosX+xpos, elemPosY+ypos, STICK_OVERLAY_CURSOR_CHAR);
+}
+
+static void osdDrawStickOverlayCursor(osd_items_e osd_item)
+{
+    rc_alias_e vertical_channel, horizontal_channel;
+
+    if (osd_item == OSD_STICK_OVERLAY_LEFT) {
+        vertical_channel = radioModes[osdConfig()->overlay_radio_mode-1].left_vertical;
+        horizontal_channel = radioModes[osdConfig()->overlay_radio_mode-1].left_horizontal;
+    } else {
+        vertical_channel = radioModes[osdConfig()->overlay_radio_mode-1].right_vertical;
+        horizontal_channel = radioModes[osdConfig()->overlay_radio_mode-1].right_horizontal;
+    }
+
+    uint8_t x_pos =  (uint8_t)scaleRange(constrain(rcData[horizontal_channel], PWM_RANGE_MIN, PWM_RANGE_MAX), PWM_RANGE_MIN, PWM_RANGE_MAX, 0, OSD_STICK_OVERLAY_WIDTH);
+    uint8_t y_pos =  (uint8_t)scaleRange(PWM_RANGE_MAX - constrain(rcData[vertical_channel], PWM_RANGE_MIN, PWM_RANGE_MAX), PWM_RANGE_MIN, PWM_RANGE_MAX, 0, OSD_STICK_OVERLAY_HEIGHT) + OSD_STICK_OVERLAY_HEIGHT - 1;
```

Comment:
`scaleRange` can invert direction, just map to `OSD_STICK_OVERLAY_HEIGHT, 0`

Also this equation is IMO incorrect, it works only when PWM_RANGE_MAX = 2*PWM_RANGE_MIN

Reply:
Scalerange: uint8_t y_pos =  (uint8_t)scaleRange(constrain(rcData[vertical_channel], PWM_RANGE_MIN, PWM_RANGE_MAX), PWM_RANGE_MIN, PWM_RANGE_MAX, OSD_STICK_OVERLAY_HEIGHT, 0) + OSD_STICK_OVERLAY_HEIGHT - 1;
Doesn't work, not sure why.

Why do you think it only works with PWM_RANGE_MAX = 2*PWM_RANGE_MIN?